### DISCUSSION TITLE
Extend deps.clj and make it optionaly setup conda python ?

### DISCUSSION THREAD

**Discussion by behrica** [borkdude/deps.clj#55] [Ideas]
I am trying to make a deeper integration of the great `libpython-clj` into Clojure, with the goal to "manage" the python environments needed for `libpython-clj` automatically "before the repl starts".

I hacked a bit on `deps.clj` and in principle it does work.

See here:
https://clojurians.zulipchat.com/#narrow/stream/215609-libpython-clj-dev/topic/libpython-clj.20in.20Container

**Comment by behrica** [comment]
This makes most sense if the `clojure` CLI command itself would gain this capability.

As there is probably no chance to get this into the official Clojure tools,  I was thinking if it could make sense to add it here. (of course completely optional) 

This would result in an improved "user experience"  with `libpython-clj`. Instead of "wondering" how to setup python for this, a user just needed to create a conda `environment.yaml` in project folder and run `clj`   (so this deps.clj)

(so this setup could be of course shared as well, as file based)

Only requirements is "conda" installed and in the PATH

**Comment by behrica** [comment]
This feature cannot be implemented as a normal clojure library, as the conda setup needed to be triggered **before** the main Clojure JVM process starts. So it needs to be part of some "entry point"

**Comment by behrica** [comment]
just wondering if this is true.... The conda environment needed o be started "before" `libpython-clj/initialize!` is called.

**Comment by behrica** [comment]
Can be done from inside `libpython-clj`, it is early enough.

**Comment by behrica** [comment]
we can close discussion.

**Reply by borkdude** [reply]
Thanks for keeping us up to date on your solution
